CONTRIBUTING.md

```diff
@@ -68,6 +68,12 @@ follow work through the issue/pull request process.
 5. **Open a pull request into `develop`.** Our main branch is `develop`, and
    this is where we want new work to be merged into. `master` is a release
    branch only.
+6. **Update `CHANGELOG.md` in your branch.** Add the pull-request number and 
+   link, and a brief, one line description under the next tag title. Make sure 
```

Comment:
Adding a PR’s number/link means an author has to create a pull request, then add a new commit again and push it again. This is very confusing, especially for beginners and also messes up with the git history since the changelog notes won’t belong to the actual commit of the change. Also, it introduces the work for maintainers since it remains first unclear if a Pull Request is ready to review or not and if someone forgot to link to its own PR, we need to wait a bit and then ask again if this has been forgotten.